conans/client/conan_api.py

```diff
@@ -202,32 +201,41 @@ def test_package(self, path=None, profile_name=None, settings=None, options=None
                                     self._client_cache.profiles_path)
 
         loader = ConanFileLoader(self._runner, self._client_cache.settings, profile)
-        conanfile = loader.load_conan(test_conanfile, self._user_io.out, consumer=True)
+        test_conanfile = loader.load_conan(test_conanfile_path, self._user_io.out, consumer=True)
 
         try:
-            # convert to list from ItemViews required for python3
-            if hasattr(conanfile, "requirements"):
-                conanfile.requirements()
-            reqs = list(conanfile.requires.items())
-            first_dep = reqs[0][1].conan_reference
-        except Exception:
-            raise ConanException("Unable to retrieve first requirement of test conanfile.py")
+            if hasattr(test_conanfile, "requirements"):
+                test_conanfile.requirements()
+        except Exception as e:
+            raise ConanException("Error in test_package/conanfile.py requirements(). %s" % str(e))
+
+        requirement = test_conanfile.requires.get(package_name)
+        if requirement:
```

Comment:
It depends on the approach we are following, if it is going to be the standard, then yes.